vortex-array/src/lib.rs

```diff
@@ -108,6 +112,14 @@ impl ArrayData {
         self.with_dyn(|a| a.nbytes())
     }
 
+    /// Return scalar value of this array if the array is constant
+    pub fn constant(&self) -> Option<Scalar> {
```

Comment:
call this `as_constant` instead, I think

Reply:
conflicted, `constant` is so short

Reply:
you got it

Reply:
yeah, but we do `as_bool_array`, etc. that returns `Option<&dyn BoolArrayTrait>`. only 3 extra characters, much clearer :D